Hp and prayer bar plugin

Malfuryent opened the pull request:
I wanted to try and managed to create my first plugin for runelite after creating the feature request :

HP bar & Pray bar feature request. #2943

The plugin adds a HP & Prayer bar into your inventory UI, which makes it easier on the eyes so you dont
have to look at the right top corner of your screen to check your hp or prayer.

![](https://gyazo.com/bf5736ee4d845e8774fc93a9ddc0fc1f.png)

This also works for the resizeable 

![](https://gyazo.com/d4aa5c72d92ffcda597dbc683c7cec05.png)

The player can also choose to remove the icons OR counters selectively.

![](https://gyazo.com/6d53f382ad58fc9b0d8c97e419eb3bf6.png)

Poison changes the color of the HP bar so you know when you got poisoned
For venom a darker green color is used.

![](https://i.gyazo.com/1deb696f324cdabb1eca2247186e605e.png)

Activating quickprayers changes the color of the prayer bar to a more lighter color.

![](https://gyazo.com/a63107b37945a1578b2a154e827e34f7.png)

The health and prayer bars now preview how much you are going to restore
when you hover over a item in the inventory.
![](https://gyazo.com/3658d43f1aa49ddbeea0f1a3b4ec18b8.png)

![](https://gyazo.com/9452bc290e36d748404747242791caee.png)

LeviSchuck commented:
Really awesome work! Could you squash these commits together? The history is a bit messy.

Malfuryent commented:
I think i did something wrong here trying to squash them together,

CasvanDongen commented on runelite-client/src/main/java/net/runelite/client/plugins/hpandprayeroverlay/hpAndPrayerOverlay.java:
```diff
@@ -0,0 +1,160 @@
+            }
+        }
+        else
+            setPosition(OverlayPosition.DYNAMIC);
```
You should use brackets on this else statement to remain consistent with the codebase.

Malfuryent commented:
Needs review, i finished writing the things i wanted for this plugin.
Maybe some other colors for the healthbar while poisoned? feel free to suggest some changes!

raqes commented:
Would be nice of it would fit exactly into the pillars depending on the interface style.

Malfuryent commented:
@raqes That's the interface styles configuration plugin right? i'll take a look to fix it :)

Malfuryent commented:
@raqes i added the check but it currently doesn't work as intended. it needs to check if InterfaceStylesPlugin is enabled, this would require injecting the InterfaceStyles plugin which i don't know how to do.
So currently you have to set the 2005 setting to 2010 if you want to move the bars back to their original spot, even if the InterfaceStyles plugin is not active.

Malfuryent commented:
Allright, i cleaned the code up, i see now that it was a big mess, should be somewhat better now.

Malfuryent commented:
Updated the main post to reflect the visual change of the PR

@raqes allright i'll take a look on how to do this.

Malfuryent commented:
Reworked the entire code so it hopefully doesn't have any duplicate code anymore.
Also updated post to reflect the visual changes to the PR
Many thanks @raqes for showing me another way to do this kind of stuff. I've learnt alot :) 👍

raqes commented on runelite-client/src/main/java/net/runelite/client/plugins/hpandprayerstatus/Viewport.java:
```diff
@@ -0,0 +1,47 @@
+@AllArgsConstructor
+enum Viewport //24
+{
+	RESIZED_BOX(WidgetInfo.RESIZABLE_VIEWPORT_OLD_SCHOOL_BOX, 161, 65, new Point(22, -8), new Point(2, - 8)),
```
Should name the group and related child in WidgetInfo instead of calling it directly with int like the first var in the enum (viewport).

Malfuryent replied:
So i should add line 37/38/39 to the net.runelite.api.widgets.WidgetInfo; file?
and then call it from there so i don't need the viewport file in the plugin?
or am i not understanding what you want me to change here?
sorry for being noob :)

raqes replied:
You basically want to replace the group and child with one a new WidgetInfo var, for which you have to make new entries in the WidgetInfo class.

raqes commented on runelite-client/src/main/java/net/runelite/client/plugins/hpandprayerstatus/HpAndStatusOverlay.java:
```diff
@@ -0,0 +1,193 @@
+			POISONED_HEALTH_LEFT = new Color(0, 145, 0);
+		}
+
+		for (Viewport viewport : Viewport.values())
```
Would be nice if you would add the logic for the viewports when an InterfaceStyle skin is on ;)

Malfuryent replied:
What would be the benefit if i did this? there are no issues when i switch between 2005 / 2010 / normal skins. So is it really needed to detect whether a skin is applied?

raqes replied:
Ah if that's the case then ignore my comment. I thought it was still the same situation as before.

Malfuryent replied:
not anymore, that was because i was directly drawing on specific coords which makes it hard to let it change dynamically, which is not the case anymore now with the rewritten code 👍

raqes commented on runelite-client/src/main/java/net/runelite/client/plugins/hpandprayeroverlay/hpAndPrayerOverlay.java:
```diff
@@ -0,0 +1,139 @@
+
+
+    @Override
+    public Dimension render(Graphics2D g) {
```
bracket next line

raqes commented on runelite-client/src/main/java/net/runelite/client/plugins/teleportnamechanges/TeleportNameChangesPlugin.java:
```diff
@@ -0,0 +1,121 @@
+@PluginDescriptor(
+	name = "Rename teleports"
+)
+public class TeleportNameChangesPlugin extends Plugin
```
Unrelated plugin to this PR

raqes commented on runelite-client/src/main/java/net/runelite/client/plugins/hpandprayerstatus/HpAndPrayerConfig.java:
```diff
@@ -0,0 +1,54 @@
+@ConfigGroup(
+		keyName = "hpandprayeroverlay",
+		name = "HP & Prayer",
+		description = "Configuration for the HP & Prayer overlay plugin"
```
Fix indendation for both ConfigGroup and ConfigItem

raqes commented on runelite-client/src/main/java/net/runelite/client/plugins/hpandprayerstatus/HpAndStatusOverlay.java:
```diff
@@ -0,0 +1,193 @@
+import java.awt.Image;
+import java.awt.image.BufferedImage;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.*;
```
Fix import order by looking at other plugins and don't use wildcards for imports.

raqes replied:
Also why are there two different plugins that are basically two different versions of the same PR?

Malfuryent replied:
Fixed that in the new commit

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/hpandprayerstatus/HpAndPrayerBarPlugin.java:
```diff
@@ -0,0 +1,64 @@
+	private  HpAndStatusOverlay overlay;
+
+	@Inject
+	public  Client client;
```
Remove injection of client and config here

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/hpandprayerstatus/HpAndPrayerBarPlugin.java:
```diff
@@ -0,0 +1,64 @@
+	@Inject
+	public HpAndPrayerConfig config;
+
+	@Inject
```
This should be @Override not @Inject

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/hpandprayerstatus/HpAndStatusOverlay.java:
```diff
@@ -0,0 +1,220 @@
+ */
+package net.runelite.client.plugins.hpandprayerstatus;
+
+import java.awt.*;
```
https://github.com/runelite/runelite/wiki/Code-Conventions import this configuration from there to format imports properly

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/hpandprayerstatus/HpAndStatusOverlay.java:
```diff
@@ -0,0 +1,220 @@
+	private SpriteManager sm;
+	private FontManager fm;
+
+	final private int HEIGHT = 250;
```
private static final?

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/hpandprayerstatus/HpAndStatusOverlay.java:
```diff
@@ -0,0 +1,220 @@
+	final private int PADDING = 1;
+
+	// Static colors
+	final Color DEPLETED = new Color(25, 25, 25, 150);
```
private static final? Basically everything here should be private static final except the private stuff, that one should be private final or local. Also, constants before instance variables

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/hpandprayerstatus/HpAndStatusOverlay.java:
```diff
@@ -0,0 +1,220 @@
+	private void renderInventoryBars(Graphics2D g)
+	{
+		//Variables
+		int max_health = c.getRealSkillLevel(Skill.HITPOINTS);
```
no snake_case in java. use camelCase

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/hpandprayerstatus/HpAndStatusOverlay.java:
```diff
@@ -0,0 +1,220 @@
+		else
+		{
+			//reset colors if not poisoned anymore.
+			HEALTH_LEFT = new Color(200, 35, 0);
```
Wait what. What is this CONSTANT naming when you are mutating it. make the variables local in method maybe?

MESLewis commented on runelite-client/src/main/java/net/runelite/client/plugins/StatusBars/StatusBarsOverlay.java:
```diff
@@ -0,0 +1,222 @@
+			counter = COUNTER_COLOR;
+		}
+		else
+			{
```
Misplaced tab

MESLewis commented on runelite-client/src/main/java/net/runelite/client/plugins/StatusBars/StatusBarsOverlay.java:
```diff
@@ -0,0 +1,222 @@
+			graphics.drawString(counter, x + 2  + padding, y + 20);
+		}
+		else
+			{
```
Misplaced tab

MESLewis commented on runelite-client/src/main/java/net/runelite/client/plugins/StatusBars/StatusBarsPlugin.java:
```diff
@@ -0,0 +1,54 @@
+import net.runelite.client.ui.overlay.Overlay;
+
+@PluginDescriptor(
+		name = "HP & Prayer statusbars"
```
Remove one tab from this line

Malfuryent commented on runelite-api/src/main/java/net/runelite/api/widgets/WidgetInfo.java:
```diff
@@ -197,6 +197,8 @@
 	RESIZABLE_VIEWPORT_EMOTES_ICON(WidgetID.RESIZABLE_VIEWPORT_OLD_SCHOOL_BOX_GROUP_ID, WidgetID.ResizableViewport.EMOTES_ICON),
 	RESIZABLE_VIEWPORT_MUSIC_ICON(WidgetID.RESIZABLE_VIEWPORT_OLD_SCHOOL_BOX_GROUP_ID, WidgetID.ResizableViewport.MUSIC_ICON),
+	RESIZABLE_VIEWPORT_INTERFACE_CONTAINER(WidgetID.RESIZABLE_VIEWPORT_OLD_SCHOOL_BOX_GROUP_ID, WidgetID.ResizableViewport.INTERFACE_CONTAINER),
+	RESIZABLE_VIEWPORT_BOTTOM_LINE_INTERFACE_CONTAINER(WidgetID.RESIZABLE_VIEWPORT_BOTTOM_LINE_GROUP_ID, WidgetID.ResizableViewport.INTERFACE_CONTAINER),
```
should be fixed now in the new commit

Malfuryent commented:
Review needed, or is everything allright?

MESLewis commented on runelite-client/src/main/java/net/runelite/client/plugins/StatusBars/StatusBarsOverlay.java:
```diff
@@ -0,0 +1,219 @@
+		return null;
+	}
+
+	private void renderInventoryBars(Graphics2D g)
```
This is just called from render, can it be inlined there?

Malfuryent replied:
i'm not sure what you want me to do hmm

Malfuryent replied:
allright, i did it and it still works. great :)

Malfuryent commented:
The bars are only visible if any interface is open ( on fixed and box mode in resizable this is always the case)
on the bottomline mode in resizable it is not visible when no interfaces are open. This seems right if a player wants to make a screenshot, or just wants to have a clean gamescreen. 
The bars will be used only whenever a player is in combat anyway in which case a interface is open ( prayer/inventory ). 

The only extra functionality that could be added is a indicator whenever you hover over a HP or prayer restoring item to see to where your hp would be filled if you ate or drank it. 
If someone wants to help add this function that would be great as i have no clue on how i would need to start on this and where to look.

but i am currently happy with the plugin as it stands.

psikoi commented:
They're looking way better now, happy to see you took everyone's feedback

Malfuryent commented:
The prayer bar now also switches color whenever the player has quick_prayers activated :)

Malfuryent commented:
Added the preview bars for prayer and health healing.
needs review.
